include/DcmTrajectoryGenerator.h

```diff
@@ -0,0 +1,391 @@
+   * @param dcmPos cartesian position of the Diverget Component of Motion.
+   * @return true / false in case of success / failure.
+   */
+  virtual bool getDcmPos(const double &t, iDynTree::Vector2& dcmPos) = 0;
+
+  /**
+   * Pure virtual method. It returns the velocity of the DCM 
+   * trajectory evaluated at time t.
+   * @param t is the trajectory evaluation time;
+   * @param dcmVel cartesian velocity of the Diverget Component of Motion.
+   * @return true / false in case of success / failure.
+   */
+  virtual bool getDcmVel(const double &t, iDynTree::Vector2& dcmVel) = 0;
+
+  /**
+   * Return true if the time t belongs to the trajectory time 
+   * domain (i.e t belongs (startTime, endTime)).
+   * @param t is the time.
+   * @return true if startTime <= t <=  endTime, false otherwise.
+   */
+  bool timeBelongsToDomain(const double &t);
+
+  /**
+   * Get the trajectory domain.
+   * @return time domain of the trajectory
+   */
+  const std::pair<double, double>& getTrajectoryDomain() const;
+};
+
+/**
+ * This class represents a coordiante (x or y) of the Divergent Component
+ * of Motion trajectory during a single support phase.
+ */
+class SingleSupportTrajectory : public GeneralSupportTrajectory
+{
+ private:
+  
+  iDynTree::Vector2 m_dcmIos; /**< Desired position of the DCM at the beginning of the step */
+  iDynTree::Vector2 m_zmp; /**< Desired position of the ZMP at the beginning of the step */
+  double m_omega; /**< Time constant of the 3D-LIPM */
+
+  /** 
+   * Absolute "init" time of the Single Support phase
+   * it is important to notice that t0 is equal to the absolute time of the previous half
+   * DS phase. 
+   * Note that m_t0 is NOT equal to the start_time (aka the time of the beginning
+   * of the SS trajectory. Because the SS trajectory is obtained under the hypotesis of
+   * no DS phase).
+   */
+  double m_t0;
+
+  /**
+   * Evaluate the position of the DCM at the beginning of the trajectory.
+   * @param nextDcmIos is the position of the DCM at the beginning of the next step;
+   * @param stepDuration is the duration of the step.
+   */
+  void setDcmIos(const iDynTree::Vector2 &nextDcmIos,
+		 const double &stepDuration);
+    
+ public:
+  /**
+   * Constructor.
+   * @param startTime is the init time of the trajectory;
+   * @param endTime is the end time of the trajectory;
+   * @param zmp is the desired position of the ZMP in the SS phase trajejctory;
+   * @param t0 is the "init" time of the SS phase;
+   * @param omega time constant of the 3D-LIPM.
+   */
+  SingleSupportTrajectory(const double &startTime,
+			  const double &endTime,
+			  const double &t0,
+			  const double &stepDuration,
+			  const double &omega,
+			  const iDynTree::Vector2 &zmp,
+			  const iDynTree::Vector2 &nextDcmIos);
+
+
+  /**
+   * DCM init of step getter.
+   * @return the position of the DCM at the beginning of SS phase.
+   */
+  const iDynTree::Vector2& getDcmIos() const;
```

Comment:
Use a longer name. ``Ios`` is not really clear. Also I would go for ``DCM`` instead of ``Dcm`` since it is an acronym, but it is more a personal taste.